src/storage/createRestStorage.ts

```diff
@@ -0,0 +1,283 @@
+	acceptPatch?: false | string;
+	contentType?: string;
+}
+
+export interface RestStorageFactory extends StorageFactory {
+	<T>(options: RestOptions<T, CrudOptions>): Storage<T, CrudOptions>;
+}
+
+export interface RestState<T> {
+	target: string;
+	createIdTarget?: string;
+	idSerializer: (ids: string[]) => string;
+	requestSerializer: (items: T[]) => string;
+	responseDeserializer: (response: string) => T[];
+	querySerializer?: (query: Query<{}, T>) => string;
+	acceptPatch: false | string;
+	contentType: string;
+	initPromise: Promise<any>;
+}
+
+function defaultIdSerializer(ids: string[]): string {
+	return `ids=${ ids.join(',') }`;
+}
+
+function defaultRequestSerializer<T>(items: T[]): string {
+	return JSON.stringify(items);
+}
+
+function defaultResponseDeserializer<T>(response: string): T[] {
+	return JSON.parse(response);
+}
+
+const instanceStateMap = new WeakMap<Storage<{}, {}>, RestState<{}>>();
+
+const JsonContentType = 'application/json';
+const JsonPatchContentType =  'application/json-patch+json';
+const MergePatchContentType = 'application/merge-patch+json';
+
+/*
+  Return the patch serializer for the first supported content type in 'Accept-Patch' header or user option.
+ */
+function getPatchSerializer(acceptPatch: string | false): (patch: Patch<{}, {}>) => string {
+	if (!acceptPatch || acceptPatch.trim().length === 0) {
+		return null;
+	}
+
+	for (let contentType of acceptPatch.split(',')) {
+		if (JsonPatchContentType === contentType.trim()) {
+			return (patch: Patch<{}, {}>) => patch.toString();
+		}
+		else if (MergePatchContentType === contentType.trim()) {
+			// TODO: dependency - merge patch is not implemented yet
+			// return (patch: Patch<{}, {}>) => patch.toMergeString();
+		}
+	}
+	return null;
+}
+
+const createRestStorage: RestStorageFactory = createBaseStorage.mixin(compose({
+
+	createId(this: Storage<{}, {}>): Promise<string> {
+		const state = instanceStateMap.get(this);
+		if (state.createIdTarget) {
+			return <Promise<any>> request.get(state.createIdTarget).then(function(response) {
+				return String(response.data);
+			});
+		}
+		else {
+			return Promise.reject(new Error('createId is not supported in a rest store.'));
+		}
+	},
+
+	fetch<V>(this: Storage<{}, {}>, query?: Query<{}, V>): Promise<V[]> {
+		const state = instanceStateMap.get(this);
+		let uri = state.target;
+		if (query) {
+			uri += '?' + query.toString(state.querySerializer);
+		}
+		return <Promise<any>> request.get(uri, {
+			headers: {
+				'Accept': state.contentType
+			}
+		}).then(function(response) {
+			return state.responseDeserializer(String(response.data));
+		});
+	},
+
+	get(this: Storage<{}, {}>, ids: string[]): Promise<{}[]> {
+		if (!ids || ids.length === 0) {
+			return Promise.reject(new Error('ids can\'t be empty.'));
+		}
+		const state = instanceStateMap.get(this);
+		let uri = state.target;
+
+		if (ids.length === 1) {
+			uri += '/' + ids[0];
+		}
+		else {
+			uri += '?' + state.idSerializer(ids);
```

Comment:
We should probably check if `state.idSerializer` exists and default to making individual get requests here if it doesn't.